ts/lib/auth/oauth2client.ts

```diff
@@ -18,17 +18,26 @@ import * as querystring from 'querystring';
 import * as request from 'request';
 
 import PemVerifier from './../pemverifier';
+import {RequestCallback, RequestError} from './../transporters';
 import AuthClient from './authclient';
+import Credentials from './credentials';
 import LoginTicket from './loginticket';
 
 const merge = require('lodash.merge');
 
+export interface GenerateAuthUrlOpts {
+  response_type?: string;
```

Comment:
Could the value of `response_type` be restricted to a set of strings so that its type could be `string1|string2|...`?